ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java

```diff
@@ -250,7 +250,8 @@ private boolean shouldReplayEvent(FileStatus dir, DumpType dumpType) throws Sema
     // repl id stored in the database/table is valid.
```

Comment:
Do we really need this method shouldReplayEvent as all txn events are idempotent when it reaches metastore and able to find it from REPL_TO_TXN_MAP table?
Yes, this may help for the case where we repeat OPEN-ABORT/COMMIT events. But, even with this method, we cannot guarantee idempotent behaviour as db/table name can be null here. So, better to push this low level before allocating writeID or copy data files during commit. Also, need to check it on each tables if a particular commit operates on multiple tables.

Reply:
remove this if check ..let it filer it for all operation

Reply:
check if some checks can be done at lower layer based on table/partition object